### DISCUSSION TITLE
Minor performance issues with menu cursor

### DISCUSSION THREAD

**Discussion by cl8n** [ppy/osu#25788] [Ideas]
I have the following problems with the menu cursor, none are unbearable but I wonder if it can be improved somehow.

- cursor fps is tied to the rest of the game, so...
  - lag spikes or any other performance drops are reflected in the cursor
  - purposely limiting fps, such as when window isn't focused, causes the cursor to become choppy
- cursor has more delay than my OS cursor (minimal, but *just enough* to throw me off a little if I'm moving my cursor in and out of the window)
- cursor either partially duplicates or partially disappears at the edges of the window
  - left outer edge: OS cursor overlaps the partially-drawn menu cursor
  - right inner edge: menu cursor gets cut off by the edge of the window, but OS cursor is still hidden

I noticed this most prominently when working with visual tests, because it's kind of a worst-case setup where I run into all of the above situations frequently

**Comment by peppy** [comment]
> cursor has more delay than my OS cursor (minimal, but just enough to throw me off a little if I'm moving my cursor in and out of the window)

This is only the case if you're running windowed. And that's by design (on the OS side). Nothing can be done there without switching to a hardware cursor.
